src/machine/machine_nrf.go

```diff
@@ -185,21 +185,19 @@ func (i2c I2C) Tx(addr uint16, w, r []byte) error {
 		}
 	}
 	if len(r) != 0 {
+		i2c.Bus.SHORTS = 1        // To trigger suspend task when a byte is received
 		i2c.Bus.TASKS_STARTRX = 1 // re-start transmission for reading
 		for i := range r {        // read each char
 			if i+1 == len(r) {
-				// The 'stop' signal must be sent before reading back the last
-				// byte, so that it will be sent by the I2C peripheral right
-				// after the last byte has been read.
-				r[i] = i2c.readLastByte()
-			} else {
-				r[i] = i2c.readByte()
+				// To trigger stop task when last byte is received, set before resume task.
+				i2c.Bus.SHORTS = 2
 			}
+			i2c.Bus.TASKS_RESUME = 1 // re-start transmission for reading
+			r[i] = i2c.readByte()
 		}
-	} else {
-		// Nothing to read back. Stop the transmission.
-		i2c.signalStop()
 	}
+	i2c.signalStop()
+	i2c.Bus.SHORTS = 0
```

Comment:
I think this should be `i2c.Bus.SHORTS &^= (nrf.TWI_SHORTS_BB_STOP << nrf.TWI_SHORTS_BB_STOP_Pos)`